pyproject.toml

```diff
@@ -127,6 +128,8 @@ types-mock = "4.0.8"
 types-PyOpenSSL = "21.0.3"
 types-PyYAML = "6.0.3"
 
+[tool.poetry.extras]
+monitoring = [ "sentry-sdk", "prometheus-client" ]
```

Comment:
I don't know if this is atomic enough (how do we do this in Synapse)?

For example, if I wanted Prometheus but not Sentry, this wouldn't let me pick and choose. But it's not a big deal and I'd probably just try and pick the same extra names as Synapse so we're not left guessing (unless we have reason to change).

Reply:
Very fair. `prometheus-client` is required in Synapse (!) and there's a `sentry` extra for `sentry-sdk`.

Reply:
That being said, in general I'd prefer that the extra's name describes the feature being offered---rather than the name of the dependency.

Reply:
In this case the name of the dependency is the same as the feature being provided, though. We can't just switch out Sentry for something else one day, since that won't provide the same interface (i.e. the operator would have to switch to this other tool we chose).

Just the same as how `postgres` would be a feature name rather than `other_database` or something.

Reply:
`metrics` or `openmetrics` might be a sensible alt to `prometheus` as a name, but I don't know how much that's interchangeable with other systems nowadays.